quality_control/models/qc_trigger_line.py

```diff
@@ -33,15 +37,16 @@ class QcTriggerProductCategoryLine(models.Model):
 
     product_category = fields.Many2one(comodel_name="product.category")
 
-    def get_trigger_line_for_product(self, trigger, product):
-        trigger_lines = super(QcTriggerProductCategoryLine,
-                              self).get_trigger_line_for_product(trigger,
-                                                                 product)
+    def get_trigger_line_for_product(self, trigger, product, partner=False):
+        trigger_lines = super(
+            QcTriggerProductCategoryLine,
+            self).get_trigger_line_for_product(trigger, product,
+                                               partner=partner)
         category = product.categ_id
         while category:
-            for trigger_line in category.qc_triggers:
-                if trigger_line.trigger.id == trigger.id:
-                    trigger_lines.add(trigger_line)
+            for trigger_line in category.qc_triggers.filtered(
+                    lambda r: r.trigger.id == trigger.id):
```

Comment:
Yeah, but this was also before knowing all the improvement in the new API recordset. Now that we know it, we can save some chars in code and instructions to execute in run time